app/ui/src/app/customizations/api-connector/api-connector-info/api-connector-info.component.ts

```diff
@@ -52,63 +60,77 @@ export class ApiConnectorInfoComponent implements OnInit {
       this.apiConnectorDataForm.get('description').setValue(description);
       this.apiConnectorDataForm.get('host').setValue(configuredProperties.host || properties.host.defaultValue);
       this.apiConnectorDataForm.get('basePath').setValue(configuredProperties.basePath || properties.basePath.defaultValue);
-      this.isDirty = true;
     }
-  }
 
-  onChange(event): void {
-    if (event.target && event.target.files) {
-      const fileList = event.target.files as FileList;
-      if (fileList.length > 0) {
-        this.iconFile = fileList[0];
-      }
-    }
-    this.isDirty = true;
-    // If component is not in "Create" mode (eg. detail page), updating
-    // any input field will automatically fire up the submit handler
     if (!this.createMode) {
-      this.onSubmit();
+      this.apiConnectorDataForm.disable();
     }
   }
 
-  onSubmit(): void {
-    if (this.apiConnectorDataForm.valid && this.isDirty) {
-      const { name, description, host, basePath } = this.apiConnectorDataForm.value;
-      const apiConnectorData = {
-        ...this.apiConnectorData,
-        configuredProperties: {
-          ...this.apiConnectorData.configuredProperties,
-          host,
-          basePath,
-        },
-        name,
-        description,
-        iconFile: this.iconFile
-      } as CustomConnectorRequest;
-
-      this.update.emit(apiConnectorData);
-    }
-    this.editControlKey = null;
+  editForm() {
+    this.enableEdit();
   }
 
-  onEditEnable(event: Event, key: string): void {
-    event.preventDefault();
-    event.stopPropagation();
-    this.editControlKey = key;
+  enableEdit() {
+    this.editMode = true;
+    this.apiConnectorDataForm.enable();
+  }
+
+  disableEdit() {
+    this.editMode = false;
+    this.apiConnectorDataForm.disable();
     this.isDirty = false;
-    setTimeout(() => this.renderer.selectRootElement(`#${key}`).focus(), 0);
   }
 
-  onEditChange(): void {
+  cancelEdit() {
+    this.disableEdit();
+    this.clearValue(this.connectorIconInput);
+    this.iconFile = undefined;
+  }
+
+  clearValue(el) {
+    el.nativeElement.value = '';
+  }
+
+  isDisabled() {
+    return !this.editMode;
+  }
+
+  onChange() {
+    if (this.connectorIconInput.nativeElement.files) {
+      const fileList = this.connectorIconInput.nativeElement.files as FileList;
+      if (fileList.length > 0) {
+        this.iconFile = fileList[0];
+      }
+      this.isDirty = true;
+    }
+  }
+
+  onEditChange() {
     this.isDirty = true;
   }
 
-  @HostListener('document:click', ['$event'])
-  private onDocumentClick(event: Event): void {
-    const element: any = event.target;
-    const elementTag = element.tagName.toLowerCase();
-    if (elementTag !== 'input' && elementTag !== 'textarea' && this.editControlKey) {
-      setTimeout(() => this.onSubmit(), 0);
+  onSubmit(): void {
+    if (this.apiConnectorDataForm.valid) {
+      if (this.isDirty) {
```

Comment:
`isDirty` can probably be replaced with looking at `this.apiConnectorDataForm.dirty`